### ISSUE TITLE
Freetds of version 1.00.86 is slow when retrieving the data

### ISSUE THREAD

**Issue by sumanthns** [FreeTDS/freetds#211]
I have a distributed setup with a rails app running in AWS, talking to a SQL server in a local datacenter via Direct Connect.

The older version of freetds installed on app server was 0.91 and everything was running fine.
I upgraded freetds alone to 1.00.86. Suddenly it added a lot of latency to db calls and overall turnaround time increased considerably.

To profile it, I ran a batch with 50 queries from the app server to db
Freetds 0.91 = Takes 0.7 seconds
Freetds 1.00.86 = Takes around 30 seconds!

Here is my freetds config.
```
[global]
        # TDS protocol version
	tds version = 4.2

	# Whether to write a TDSDUMP file for diagnostic purposes
	# (setting this to /tmp is insecure on a multi-user system)
;	dump file = /tmp/freetds.log
;	debug flags = 0xffff

	# Command and connection timeouts
;	timeout = 10
;	connect timeout = 10

	# If you get out-of-memory errors, it may mean that your client
	# is trying to allocate a huge buffer for a TEXT field.
	# Try setting 'text size' to a more reasonable limit
	text size = 64512
	encryption = require

	# If you experience TLS handshake errors and are using openssl,
	# try adjusting the cipher list (don't surround in double or single quotes)
	# openssl ciphers = HIGH:!SSLv2:!aNULL:-DH

# A typical Sybase server
[egServer50]
	host = symachine.domain.com
	port = 5000
	tds version = 5.0

# A typical Microsoft server
[egServer70]
	host = ntmachine.domain.com
	port = 1433
	tds version = 7.0
```

These are the things I have tried already, but to no avail
1. Turn off logs as suggested in [userguide](http://www.freetds.org/userguide/seemtooslow.htm)
2. Turn off encryption

I also noticed the same behavior on freetds 1.00.27 as well.
But the moment I rolled back to 0.91, everything is good

**Comment by freddy77** [comment]
Pretty weird.
Which system are you using? Linux?
A TDSDUMP would be helpful.
Are you using the same TDS protocol?
The difference is quite massive, surely something wrong. I would check if the data exchanged are more or less the same, the round-trip count and network packet timing.
Which kind of query are you sending? Normal ones or prepared?
Which library are you using?

**Comment by sumanthns** [comment]
I am using the same TDS protocol, 4.2. The data exchanged is exactly the same. Using https://github.com/rails-sqlserver/tiny_tds to connect from rails app

**Comment by nicomen** [comment]
This is the same experience I have using perl DBI::ODBC + freetds, 0.91 is fast, 1.00.82 slow. Not sure how to do a TDSDUMP, but I'll gladly provide one with both versions if I know how.

**Comment by freddy77** [comment]
I never encountered this issue. Are you using Linux or another OS? Are you using encrypted connection?

**Comment by freddy77** [comment]
Can I have access to a machine with such issues?

**Comment by freddy77** [comment]
I'll close the ticket if I don't get a reply in a month.

**Comment by freddy77** [comment]
I think I got the issue. Has to do with the way CORK flag is used and if the queue is full. In this case kernel if not able to flush when CORK is set but being reset just after won't allow the kernel to flush data when the queue became free. This is due to setting only CORK but not NODELAY.

**Comment by freddy77** [comment]
Fixed in 1.00.101 (package expected tomorrow)
